proposals/utils/validate_proposal.py

```diff
@@ -1,322 +1,82 @@
-                proposal.wmo,
-            )
-
-    # Now we're done for pre assessment proposals
-    if proposal.is_pre_assessment:
-        return forms
-
-    forms["study_start"] = (
-        StudyStartForm,
-        reverse("proposals:study_start", args=[proposal.pk]),
-        _("Eén of meerdere trajecten?"),
-        proposal,
-    )
-
-    for study in proposal.study_set.all():
-        key_base = "study_{}".format(study.pk)
-
-        start_key = "{}_start".format(key_base)
-        forms[start_key] = (
-            StudyForm,
-            reverse("studies:update", args=[study.pk]),
-            _("De deelnemers (traject {})").format(study.order),
-            study,
-        )
-
-        design_key = "{}_design".format(key_base)
-        forms[design_key] = (
-            StudyDesignForm,
-            reverse("studies:design", args=[study.pk]),
-            _("De onderzoekstype(n) (traject {})").format(study.order),
-            study,
-        )
-
-        end_key = "{}_end".format(key_base)
-        forms[end_key] = (
-            StudyEndForm,
-            reverse("studies:design_end", args=[study.pk]),
-            _("Trajectoverzicht (traject {})").format(study.order),
-            study,
-        )
-
-        if study.has_intervention:
-            intervention_key = "{}_intervention".format(key_base)
-            if hasattr(study, "intervention"):
-                forms[intervention_key] = (
-                    InterventionForm,
-                    reverse("interventions:update", args=[study.intervention.pk]),
-                    _("Het interventieonderzoek (traject {})").format(
-                        study.order,
-                    ),
-                    study.intervention,
-                )
-            else:
-                forms[intervention_key] = (
-                    InterventionForm,
-                    reverse("interventions:create", args=[study.pk]),
-                    _("Het interventieonderzoek (traject {})").format(
-                        study.order,
-                    ),
-                    None,
-                )
-
-        if study.has_observation:
-            observation_key = "{}_observation".format(key_base)
-            if hasattr(study, "observation"):
-                forms[observation_key] = (
-                    ObservationForm,
-                    reverse("observations:update", args=[study.observation.pk]),
-                    _("Het observatieonderzoek (traject {})").format(
-                        study.order,
-                    ),
-                    study.observation,
-                )
-            else:
-                forms[observation_key] = (
-                    ObservationForm,
-                    reverse("observations:create", args=[study.pk]),
-                    _("Het observatieonderzoek (traject {})").format(
-                        study.order,
-                    ),
-                    None,
-                )
+def validate_sessions_tasks(study):
```

Comment:
I would really really prefer if we could somehow have this be part of the stepper/checker infrastructure, so that all we had to do was run a `stepper.get_all_errors` to get list of errors with titles and URLs.

For example, calling this code from the `SessionsChecker`. Or otherwise a `ModelFormItem` for each of the task pages that is hidden but does collect errors.

That would keep everything in one place and allow us to turn the Sessions item yellow if its subpages had errors.